doc/rst/developer/bestPractices/MemoryProfiling.rst

```diff
@@ -0,0 +1,85 @@
+.. _best-practices-memory-profiling:
+
+Memory Profiling
+================
+
+When using a non-standard memory allocator such as jemalloc, the allocator may
+provide profiling capabilities that can be useful for understanding memory
+usage patterns. For Chapel programs, this can be used to augment the existing
+Chapel runtime's memory tracking capabilities. For the Chapel compiler, this
+can be used to understand the memory usage of the compiler itself.
+
+Profiling the compiler with jemalloc
+------------------------------------
+
+To profile the compiler with jemalloc, make sure you build the compiler
+with the following environment variables set:
+
+.. code-block:: bash
+
+   export CHPL_HOST_MEM=jemalloc
+   export CHPL_HOST_JEMALLOC="bundled"
+   export CHPL_JEMALLOC_ENABLE_PROFILING=1
+
+Then the compiler can be run as follows:
+
+.. code-block:: bash
+
+   MALLOC_CONF='prof:true,prof_prefix:jeprof.out,prof_final:true' \\
+     chpl myProgram.chpl
+
+The value of ``prof_prefix`` is the prefix for the output files.
+
+It may be useful to pass ``--no-compiler-driver`` to the compiler. Without this
```

Comment:
This feels `.. note::` worthy.  If it was in the Hugo blog, I'd want it off to the side but I'm not sure how to do that in Sphinx.  So I think maybe `.. note::` is the next best thing?  I'm also okay with leaving it in the main body, it just feels like a bit of a sidebar